tests/test_regression_against_cases.py

```diff
@@ -46,3 +47,27 @@ def test_regression_against_case_with_update(case: Path):
 
     dataset, reference_dataset = xr.align(dataset, reference_dataset)
     assert_allclose(dataset, reference_dataset)
+
+
+@pytest.mark.regression
+@pytest.mark.filterwarnings("ignore:Not all input parameters were used")
+def test_regression_against_case_with_repeated_update():
+    case = CASES_DIR / "SPARC_PRD" / "input.yaml"
+
+    input_parameters, algorithm, _, _ = read_case(case)
+    input_parameters["average_electron_density"] = input_parameters["average_electron_density"][::5]
+    input_parameters["average_electron_temp"] = input_parameters["average_electron_temp"][::5]
+
+    dataset = xr.Dataset(input_parameters)
+
+    first_run = algorithm.update_dataset(dataset)
+    second_run = algorithm.update_dataset(first_run)
```

Comment:
What do you think about adding `import copy` and doing the following.
```suggestion
    first_run = algorithm.update_dataset(dataset)
    first_run_copy = copy.deepcopy(first_run)
    second_run = algorithm.update_dataset(first_run_copy)
```

This makes clear that `first_run` won't get modified by `update_dataset` w/o knowing the details of `update_dataset`. We could even assert that `first_run` and `first_run_copy` are identical after the call to ensure `update_dataset` doesn't modify `first_run`.